src/index.js

```diff
@@ -281,13 +281,19 @@ function sanitizeHtml(html, options, _recursing) {
                 parsed = url.parse(value, false, true);
                 var isRelativeUrl = parsed && parsed.host === null && parsed.protocol === null;
                 if (isRelativeUrl) {
-                  // default value of allowIframeRelativeUrls is true unless allowIframeHostnames specified
+                  // default value of allowIframeRelativeUrls is true
+                  // unless allowedIframeHostnames or allowedIframeDomains specified
                   allowed = has(options, "allowIframeRelativeUrls")
-                    ? options.allowIframeRelativeUrls : !options.allowedIframeHostnames;
-                } else if (options.allowedIframeHostnames) {
-                  allowed = options.allowedIframeHostnames.find(function (hostname) {
+                    ? options.allowIframeRelativeUrls
+                    : (!options.allowedIframeHostnames && !options.allowedIframeDomains);
+                } else if (options.allowedIframeHostnames || options.allowedIframeDomains) {
+                  var allowedHostname = (options.allowedIframeHostnames || []).find(function (hostname) {
                     return hostname === parsed.hostname;
                   });
+                  var allowedDomain = (options.allowedIframeDomains || []).find(function(domain) {
+                    return parsed.hostname.endsWith(domain);
```

Comment:
Need to make sure it's either matching the entire string, or ending with the domain preceded by a dot. Otherwise you can fake it out with a newly registered domain name, for instance if youtube.com is allowed you can register fakeyoutube.com and it will pass.

Reply:
I appreciate the effort, sorry for the nitpicking but it we do need to secure it.

Reply:
Oh, you are deadly right!

Reply:
I'll cover this case with a test.